nemo/collections/asr/models/ctc_models.py

```diff
@@ -37,17 +44,23 @@ class EncDecCTCModel(ASRModel):
     def __init__(self, cfg: DictConfig, trainer: Trainer = None):
         if 'cls' not in cfg:
             # This is for Jarvis service. Adding here for now to avoid effects of decorators
-            OmegaConf.set_struct(cfg, False)
-            cfg.cls = 'nemo.collections.asr.models.EncDecCTCModel'
-            OmegaConf.set_struct(cfg, True)
+            # Assuming cfg will have the config structure CONSISTENT with the modules, i.e.: {cls: ..., params: ...}
+            config = {"cls": 'nemo.collections.asr.models.EncDecCTCModel'}
+            if "params" in cfg:
+                config["params"] = cfg.params
+            else:
+                config["params"] = cfg
```

Comment:
Why are we pushing the entire cfg into the "params" namespace here? This will break optimizers and schedulers cause they assume everything is run under the model.optim namespace.

Reply:
On top of this, there is no good reason to push everything into config. You could easily simply add the cfg.cls as I did above with struct overriding and treat the rest of the namespace as part of `cfg` itself - this is easier to interpret, and still serves the same purpose.

Reply:
Honestly, I did this change just to be consistent, as described in the comment above:
https://github.com/NVIDIA/NeMo/blob/c7988ef531fc0092004e516347feac17ff4e62a4/nemo/collections/asr/models/ctc_models.py#L47

I have went through @okuchaiev examples/tests and he assumed that his:
https://github.com/NVIDIA/NeMo/blob/c7988ef531fc0092004e516347feac17ff4e62a4/tests/collections/asr/test_asr_ctcencdec_model.py#L25

So I just followed. It doesn't even mean that I agree with this - I have just followed the same pattern...

Reply:
Yes and that pattern doesnt work with all of optim namespace. I dont understand why a PR to refactor tests from unittest to pytest has gone ahead and broken a good chunk of other stuff unrelated to tests.

Reply:
@titu1994 regarding optimizers: this is important, are you sure?

I have just relied on our tests, and _believed_ that if tests are passing, then I am not breaking anything:
http://10.110.42.104:8080/blue/organizations/jenkins/NeMo/detail/PR-871/12/pipeline